### DISCUSSION TITLE
Can't activate in Visual Studio, always print NotSignedIn

### DISCUSSION THREAD

**Discussion by Yi-Frank** [community/community#13947] [Copilot Conversations] [answered]
![image](https://user-images.githubusercontent.com/24606096/160978189-ba7d73bd-2905-4933-8b46-193779b2764f.png)
```
GitHub Copilot 1.10.0.2 (v1.10.0.2@f18eaf35a)
GitHub Copilot Agent v1.10.0
NotSignedIn
NotSignedIn
NotSignedIn
NotSignedIn
NotSignedIn
```

Whether I choose "OK" or "Cancel", it always shows "NotSignedIn"....and never open the browser...
I try Administrator power and Users power to run VS, but it makes no difference...
but it can install and work in jetbrains and vscode....

> Microsoft Visual Studio Professional 2022
> Version 17.1.2
> VisualStudio.17.Release/17.1.2+32319.34

**Comment by Yi-Frank** [comment, marked as answer]
solved, that is certificate problem....

**Reply by redcoreit** [reply]
I have the same issue. What cert exactly?
What was the solution? Can you provide some details?

**Reply by RodherDigital** [reply]
Exactly, how did you solve?

**Reply by Yi-Frank** [reply]
@redcoreit @Dekklezz I used the proxy network before, and then I closed the proxy network.

The reason for this problem is that the extension doesn't load the CA certificate correctly, so it cannot use proxy network access, but VPN can be used.

**Reply by redcoreit** [reply]
Thank you @Yi-Frank . 
Sadly I cant change or close corpo proxy :D 

Do you think copilot should work through a properly configured proxy?
I have no other cert related issue tho.

**Reply by Yi-Frank** [reply]
@redcoreit It's nothing. 
Yes, I think copilot should load the CA certificate form system, or provides the option to turn off SSL authentication. 
but I don't know when it will be updated...

**Reply by zanthus** [reply]
I connected my work PC to my mobile hotspot and allowed copilot to login. Then I reconnected to my corporate LAN and all is well :)

**Reply by StarsDragonT1** [reply]
![屏幕截图 2023-08-21 222219.png](https://github.com/community/community/assets/130347888/c7deeed5-ca55-4a73-bbee-b49fa9a564c2)

Sorry to interrupt, but I think I'm having some problems. The plugin keeps displaying "NotSignedIn". Being a student, I'm having trouble understanding the bugs below.

![屏幕截图 2023-08-21 222333.png](https://github.com/community/community/assets/130347888/ee9c4fa0-fde3-435e-9e9e-11563854df80)

**Comment by maroonedsia** [comment]
Is there a proper solution to this? I have the corporate proxy problem too. Here is the start of the error message in Output => Github Copilot:

```
[INFO] [auth] [2023-08-22T20:01:26.933Z] Invalid copilot token: missing token: 404 Not Found
Auth Status: NotAuthorized
Auth Status: NotSignedIn
[Error] An error occurred for 'CreateCopilotService': StreamJsonRpc.RemoteInvocationException: Request signInWithGithubToken failed with message: self signed certificate in certificate chain
   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__151`1.MoveNext()

```   
There is a discussion here too, but no solution for VS2022: https://stackoverflow.com/questions/71367058/self-signed-certificate-in-certificate-chain-on-github-copilot
